internal/broker/broker_test.go

```diff
@@ -827,6 +831,147 @@ func TestConcurrentIsAuthenticated(t *testing.T) {
 	}
 }
 
+func TestIsAuthenticatedAllowedUsersConfig(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		allowedUsers          map[string]struct{}
+		owner                 string
+		ownerAllowed          bool
+		allUsersAllowed       bool
+		firstUserBecomesOwner bool
+
+		allowedUsernames   []string
+		unallowedUsernames []string
+	}{
+		"All_users_are_allowed": {
+			allUsersAllowed:  true,
+			allowedUsernames: []string{"u1", "u2"},
+		},
+		"Only_owner_allowed": {
+			ownerAllowed:       true,
+			owner:              "machine_owner",
+			allowedUsernames:   []string{"machine_owner"},
+			unallowedUsernames: []string{"u1", "u2"},
+		},
+		"No_users_allowed": {
+			ownerAllowed:       true,
+			owner:              "",
```

Comment:
```suggestion

```

Reply:
If I remove this it will be like testing the following:
```
owner = 
allowed_users =
```

But I am trying to test:
```
owner = 
allowed_users = OWNER
```
which is equivalent to:
```
owner = 
```

Should I add an extra test to test what you suggest?

Reply:
> But I am trying to test:
> ```
> owner = 
> allowed_users = OWNER
> ```

Isn't that already tested by https://github.com/ubuntu/authd-oidc-brokers/blob/9cc259e0d8bc26e125009b4d934292cbdcbdbc55/internal/broker/broker_test.go#L861-L866

Reply:
Ah no it's not, because `firstUserBecomesOwner` is `false` when you explicitly set owner to an empty value.

Reply:
How about we use these two test cases:

```
"No_users_allowed": {
	unallowedUsernames: []string{"u1", "u2", "machine_owner", "u3"},
},
"No_users_allowed_when_owner_is_allowed_but_not_set": {
	ownerAllowed:       true,
	unallowedUsernames: []string{"u1", "u2", "machine_owner", "u3"},
},
```